packages/builtin/string.pony

```diff
@@ -207,7 +207,7 @@ class val String is (Seq[U8] & Comparable[String box] & Stringable)
     null terminator.
     """
     if _alloc <= len then
-      _alloc = len.min(len.max_value() - 1) + 1
+      _alloc = (len + 1).next_pow2().min(len.max_value() - 1)
```

Comment:
This line has some problems.

If `len` is more than half the `max_value`, then `(len + 1).next_pow2()` will be `0`.